src/Expander.hs

```diff
@@ -551,52 +553,52 @@ initializeKernel = do
               v <- freshMacroVar
               bind b $ EIncompleteMacro v theName macroDest
               return (theName, v, macroDest)
-            linkDecl dest $ DefineMacros macros
-            nowValidAt pdest (SpecificPhase p)
+            linkOneDecl dest $ DefineMacros macros
+            linkDeclValidity vp (ScopeSet.singleScopeAtPhase sc p)
         )
       , ( "example"
-        , \ sc dest pdest stx -> do
+        , \ sc dest vp stx -> do
             p <- currentPhase
             Stx _ _ (_ :: Syntax, expr) <- mustHaveEntries stx
             exprDest <- liftIO $ newSplitCorePtr
             sch <- liftIO newSchemePtr
-            linkDecl dest (Example (view (unSyntax . stxSrcLoc) stx) sch exprDest)
+            linkOneDecl dest (Example (view (unSyntax . stxSrcLoc) stx) sch exprDest)
             t <- inTypeBinder do
               t <- Ty . TMetaVar <$> freshMeta
               forkExpandSyntax (ExprDest t exprDest) (addScope p expr sc)
               return t
             forkGeneralizeType exprDest t sch
-            nowValidAt pdest (SpecificPhase p)
+            linkDeclValidity vp (ScopeSet.singleScopeAtPhase sc p)
         )
       , ( "import"
          -- TODO Make import spec language extensible and use bindings rather than literals
-        , \sc dest pdest stx -> do
+        , \sc dest vp stx -> do
             Stx scs loc (_ :: Syntax, toImport) <- mustHaveEntries stx
             spec <- importSpec toImport
             modExports <- getImports spec
             flip forExports_ modExports $ \p x b -> inPhase p do
               imported <- addRootScope' $ addScope' (Stx scs loc x) sc
               addImportBinding imported b
-            linkDecl dest (Import spec)
-            nowValidAt pdest AllPhases
+            linkOneDecl dest (Import spec)
+            linkDeclValidity vp (ScopeSet.singleUniversalScope sc)
         )
       , ( "export"
-        , \_sc dest pdest stx -> do
+        , \sc dest vp stx -> do
             Stx _ _ (_, protoSpec) <- mustBeCons stx
             exported <- exportSpec stx protoSpec
             p <- currentPhase
             es <- getExports exported
             modifyState $ over expanderModuleExports $ (<> es)
-            linkDecl dest (Export exported)
-            nowValidAt pdest (SpecificPhase p)
+            linkOneDecl dest (Export exported)
+            linkDeclValidity vp (ScopeSet.singleScopeAtPhase sc p)
```

Comment:
The `Scope` was previously unused, presumably because `export` doesn't bind anything. So I think that the `nowValidAt pdest (SpecificPhase p)` was filling its `DeclValidityPtr` with a dummy phase, but that I could now fill it with an empty `ScopeSet` instead?

Reply:
I changed it to return `mempty`. Later, I changed the `declPrim` contract from "fill the DeclValidityPtr with a ScopeSet listing the newly-bound Scopes" to "receive a ScopeSet, add the newly-bound Scopes to it, then fill the DeclValidityPtr with the result", so now this instead returns the input ScopeSet unmodified.

Reply:
Now returning `mempty` again.